optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/bendablelong/BendableLongScoreInliner.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.buildin.bendablelong;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore;
+import org.optaplanner.core.impl.score.inliner.LongWeightedScoreImpacter;
+import org.optaplanner.core.impl.score.inliner.ScoreInliner;
+
+public class BendableLongScoreInliner extends ScoreInliner<BendableLongScore> {
+
+    private long[] hardScores;
+    private long[] softScores;
+
+    public BendableLongScoreInliner(boolean constraintMatchEnabled, int hardLevelsSize, int softLevelsSize) {
+        super(constraintMatchEnabled);
+        hardScores = new long[hardLevelsSize];
+        softScores = new long[softLevelsSize];
+    }
+
+    @Override
+    public LongWeightedScoreImpacter buildWeightedScoreImpacter(BendableLongScore constraintWeight) {
+        if (constraintWeight.equals(BendableLongScore.zero(hardScores.length, softScores.length))) {
+            throw new IllegalArgumentException("The constraintWeight (" + constraintWeight + ") cannot be zero,"
+                    + " this constraint should have been culled during node creation.");
+        }
+        Integer singleLevel = null;
```

Comment:
This code probably deserves a couple comments. Took me 5 minutes to figure out how and why this ever becomes null and non-null, and what needs to happen afterwards.

The need for comments could be slightly reduced, if in the condition on line 54, the different branches were converted into descriptively named methods.

Reply:
Deferring to https://issues.jboss.org/browse/PLANNER-1584 - I suspect that PLANNER-1578 might influence this.